lib/portlayer/exec/container_cache.go

```diff
@@ -85,6 +86,25 @@ func (conCache *containerCache) Containers(states []State) []*Container {
 	return containers
 }
 
+func (conCache *containerCache) References() []types.ManagedObjectReference {
+	conCache.m.RLock()
+	defer conCache.m.RUnlock()
+	// cache contains 2 items for each container
+	capacity := len(conCache.cache) / 2
```

Comment:
The duplication here is minimal, but it's presumably something we don't want long-term.

Could you add a `// TODO` here (and in any other places you'd expect to be changed as a part of #7558)? I think that would help ensure that we don't miss things.

Reply:
Yeah - I considered a `Map` base function to perform iteration and the filtering of entries to just the IDs, then reworking Containers and References as functions to be mapped to each entry by that based function, but is unnecessary complexity for the first pass.